src/Data-layer.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from "react";
+import { getAllVideos } from "./api-calls/index";
+import { useVideoDataFromServer } from "./context/video-context";
+
+const DataLayer = ({ children }) => {
+  const { videoDispatch } = useVideoDataFromServer();
+
+  const getVideosFromServer = async () => {
+    const serverVideos = await getAllVideos();
+    videoDispatch({
+      type: "GET_ALL_VIDEOS_FROM_SERVER",
+      payload: {
```

Comment:
better to make another file app.js and map all the reducer types, then call them here, would reduce possibility of typos
eg `const actionTypes = {ACTION_TYPE:"ACTION_TYPE_IN_STRING"}`

Not recommending  to change it in this project but from next project try doing this as its the best practice, also to avoid error as these types are being used many times in code.

Reply:
I'll keep that in mind when I work on the next project
thanks Nikhil  for reviewed